src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/mappers/LogicalPortMapper.java

```diff
@@ -0,0 +1,53 @@
+/* Copyright 2021 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.switchmanager.mappers;
+
+import org.openkilda.messaging.info.switches.LogicalPortInfoEntry;
+import org.openkilda.messaging.info.switches.LogicalPortType;
+import org.openkilda.messaging.model.grpc.LogicalPort;
+import org.openkilda.model.LagLogicalPort;
+import org.openkilda.model.PhysicalPort;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
+
+import java.util.Optional;
+
+@Mapper(imports = {LogicalPortType.class})
+public interface LogicalPortMapper {
+
+    LogicalPortMapper INSTANCE = Mappers.getMapper(LogicalPortMapper.class);
+
+    @Mapping(target = "physicalPorts", source = "portNumbers")
+    @Mapping(target = "expected", ignore = true)
+    @Mapping(target = "actual", ignore = true)
+    LogicalPortInfoEntry map(LogicalPort logicalPort);
+
+    @Mapping(target = "type", expression = "java(LogicalPortType.LAG)")
+    @Mapping(target = "expected", ignore = true)
+    @Mapping(target = "actual", ignore = true)
+    LogicalPortInfoEntry map(LagLogicalPort logicalPort);
+
+    default Integer map(PhysicalPort physicalPort) {
```

Comment:
Why not to use abstract class intead? In this case you can not use this terrible mock syntax.